catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java

```diff
@@ -2407,6 +2407,8 @@ protected ResourceResponse getResource(ResourceRequest resourceRequest, boolean
                             e);
                 }
             }
+
+            resourceResponse.getProperties().put(Constants.METACARD_PROPERTY, metacard);
```

Comment:
Why don't we add the metacard to the request instead of the response, then it will be available for the entire chain, as opposed to just the response section.